proposals/skip-localsinit.md

```diff
@@ -0,0 +1,91 @@
+The bug should be fixed and users should get a more documented and reliable way of suppressing the flag. 
+
+## Detailed design
+
+Allow specifying `System.Runtime.CompilerServices.SkipLocalsInitAttribute` as a way to tell the compiler to not emit `localsinit` flag.
+ 
+The end result of this will be that the locals may not be zero-initialized by the JIT, which is in most cases unobservable in C#.  
+In addition to that `stackalloc` data will not be zero-initialized. That is definitely observable, but also is the most motivating scenario.
+
+Permitted and recognized attribute targets are: `Method`, `Property`, `Module`, `Class`, `Struct`, `Interface`, `Constructor`. However compiler will not require that attribute is defined with the listed targets nor it will care in which assembly the attribute is defined. 
+
+When attribute is specified on a container (`class`, `module`, containing method for a nested method, ...), the flag affects all methods contained within the container.
+
+Synthesized methods "inherit" the flag from the logical container/owner. 
+
+The flag affects only codegen strategy for actual method bodies. I.E. the flag has no effect on abstract methods and is not propagated to overriding/implementing methods.
+
+This is explicitly a **_compiler feature_** and **_not a language feature_**.  
+Similarly to compiler command line switches the feature controls implementation details of a particular codegen strategy and does not need to be required by the C# spec.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+- Old/other compilers may not honor the attribute.
+Ignoring the attribute is compatible behavior. Only may result in a slight perf hit.
+
+- The code without `localinits` flag may trigger verification failures.
+Users that ask for this feature are generally unconcerned with verifiability. 
+ 
+- Applying the attribute at higher levels than an individual method has nonlocal effect, which is observable when `stackalloc` is used. 
+Yet, this is the most requested scenario.
+
+## Alternatives
+[alternatives]: #alternatives
+
+- omit `localinits` flag when method is declared in `unsafe` context. 
+That could cause silent and dangerous behavior change from deterministic to nonditerministic in a case of `stackalloc` .
+
+- omit `localinits` flag always.
+Even worse than above.
+
+- omit `localinits` flag unless `stackalloc` is used in the method body.
+Does not address the most requested scenario and may turn code unverifiable with no option to revert that back.
+
+## Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Should the attribute be actually emitted to metadata? 
```

Comment:
Also, are we going to add the attribute to public surface so that folks can use it without local definition? (I assume that local private per-assembly definition would be possible as well.)

Reply:
Compiler does not care about the identity of attribute types, so it can be defined anywhere including an internal type in source.  
Even though it is not a requirement, adding the attribute to public surface might be useful.

I did not mentioned that intentionally, to not get into matters  of nuget packages, reference assemblies, netstandard v-next and the like. - To defer to people who know better what treatment would be appropriate in this case. 
I should add this to "unresolved questions" though.